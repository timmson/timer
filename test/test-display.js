const React = require("react");
const PropTypes = require("prop-types");
const renderer = require("react-test-renderer");

const Display = require("../src/display");

class MockDisplayValue extends React.Component {

	static propTypes = {
		value: PropTypes.string,
		style: PropTypes.object
	};

	constructor(props, context) {
		super(props, context);
	}


	render() {
		return <span style={this.props.style}> {this.props.value}</span>;
	}
}

describe("Display should", () => {

	test("return stopped timer", () => {
		const component = renderer.create(
			<Display displayValue={MockDisplayValue} status={Display.STOPPED} value={[0, 1]}/>
		);
		expect(component.toJSON()).toMatchSnapshot();
		component.unmount();
	});

	test("return started timer", () => {
		const component = renderer.create(
			<Display displayValue={MockDisplayValue} status={Display.STARTED} value={[0, 1]}/>
		);
		expect(component.toJSON()).toMatchSnapshot();
		component.unmount();
	});

	test("return alerted timer", () => {
		const component = renderer.create(
			<Display displayValue={MockDisplayValue} status={Display.ALERTED} value={[0, 0]}/>
		);
		expect(component.toJSON()).toMatchSnapshot();
		component.unmount();
	});

	test("contain properties", () => {
		expect(Display.propTypes.status).toEqual(PropTypes.string);
		expect(Display.propTypes.value).toEqual(PropTypes.array);
		expect(Display.propTypes.onClick).toEqual(PropTypes.func);
	});

});