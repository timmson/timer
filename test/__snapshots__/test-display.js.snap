// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should return alerted timer 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="timer alert"
    >
      00
      <span
        style={Object {}}
      >
        :
      </span>
      00
    </div>
  </div>
</div>
`;

exports[`Display should return started timer 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="timer normal"
    >
      01
      <span
        style={
          Object {
            "animation": "blinker 1s linear infinite",
          }
        }
      >
        :
      </span>
      00
    </div>
  </div>
</div>
`;

exports[`Display should return stopped timer 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="timer normal"
    >
      01
      <span
        style={Object {}}
      >
        :
      </span>
      00
    </div>
  </div>
</div>
`;
